chapters/intro-computer-architecture/binary-hex/reading/README.md

```diff
@@ -0,0 +1,142 @@
+If bits at the same positions in operands have equal values, the resulting bit is 0;
+otherwise, it's 1.
+
+Example:
+```text
+0b1001 XOR 0b0111 = 0b1110
+```
+
+#### Logical Shifts
+
+Logical shifts left/right involve moving each bit by one position.
+Since the result must be on the same number of bits as the initial value, the first bit is lost, and the empty space is filled with a 0 bit.
+
+![Left Logical Shift](../media//left-logical-shift.svg)
+
+![Right Logical Shift](../media//right-logical-shift.svg)
+
+For explanations related to bitwise operations in C, refer to the guide at [Bitwise Operators in C](https://www.geeksforgeeks.org/bitwise-operators-in-c-cpp/).
+
+## Hexadecimal System
+
+In the **hexadecimal system** (base 16), values are represented as a string of characters from '0' to '9' or 'a' to 'f'.
+A byte consists of two such characters, so each character corresponds to a group of 4 bits (a nibble).
+
+### Conversion from Decimal to Binary/Hexadecimal
+
+- Divide the number successively by the base number (2 or 16) and keep the remainders.
+- When the quotient of the division becomes 0, write down the remainders in reverse order.
+- In the case of base 16, when the remainder is greater than 9, letters a-f are used (`0xa = 10`, `0xf = 15`).
+
+#### Example: Conversion of the number 0xD9B1 to decimal
+
+$$ \texttt{0xD9B1} = 1 \cdot 16 ^ 0 + 11 \cdot 16 ^ 1 + 9 \cdot 16 ^ 2 + 13 \cdot 16 ^ 3 = 55729 $$
+
+### Conversion between Binary and Hexadecimal
+
+As mentioned earlier, a digit in a hexadecimal number corresponds to a group of 4 bits (a nibble).
+Therefore, to convert a number from hexadecimal to binary, it's sufficient to transform each digit into the equivalent 4-bit group.
+
+#### Example: Conversion of the number 0xD9B1 to binary
+
+- `0x1 = 0b0001`
+- `0xB = 0b1011`
+- `0x9 = 0b1001`
+- `0xD = 0b1101`
+
+Thus, the resulting binary number is `0b1101100110110001`.
+
+The reverse operation, conversion from binary to hexadecimal, can be done by converting each group of 4 bits into the corresponding digit in hexadecimal.
+
+### Use of Base 16 Representation
+
+The hexadecimal system is used to represent memory addresses and to visualize data in a more interpretable way than a sequence composed only of 0s and 1s.
+The image below provides an example in this regard:
+
+![Memory Map](../media/hex-view.png)
+
+(Image taken from [Digital Detective](https://www.digital-detective.net/))
+
+## Representation of Data Types
+
+In a computer's memory, a value is stored on a fixed number of bits.
+Depending on the architecture, each processor can access a maximum number of bits in a single operation, which represents the word size.
+
+The sizes of common data types used in C are dependent on both the processor and the platform on which the program was compiled (operating system, compiler).
+The table below presents the sizes of data types on a 32-bit architecture processor, when the program is compiled using gcc under Linux.
```

Comment:
While it's good to be precise when presenting information (as to not spread misinformation), this sentence does leave the reader wondering, "but what about on 64 bit?", "what about other compilers", "what about other operating systems?". For the second and third questions, I'm not sure there are any major differences between OSes/compilers in regards to how something like `int`'s size gets treated. I would personally drop the last part and replace it with something like:
> The table below presents the sizes of data types on a 32-bit architecture processor. Most major C compilers and operating systems use these sizes, there are rare exceptions.

And in regards to question 1: I'm not sure that this lab would be appropriate to explore the differences between 32 bit and 64 bit, but I do strongly believe that an answer, however brief it may be, should always be included. A note after the table could suffice:

> Note: In terms of data types, the main difference between 32 bit systems and 64 bit systems is the size of the pointer and of the `long`. Both are 8 bytes long on 64 bit machines.

On the other hand, I would understand why we would prefer down the road to not go into much (if any) detail regarding 64 bit.

Reply:
The size of these data types are specified by the C standard and compilers have nothing to do with them. The only difference between 32-bit and 64-bit architectures are `long`s, pointers and types that are directly related to the architecture, such as `size_t` and `ssize_t`. I've added a mention about this.